packages/next/client/with-router.tsx

```diff
@@ -16,17 +16,9 @@ export default function withRouter<
   C = NextPageContext
 >(
   ComposedComponent: NextComponentType<C, any, P>
-): React.ComponentClass<ExcludeRouterProps<P>> {
-  class WithRouteWrapper extends React.Component<ExcludeRouterProps<P>> {
-    static displayName?: string
-    static getInitialProps?: any
-    static contextType = RouterContext
-
-    context!: React.ContextType<typeof RouterContext>
-
-    render() {
-      return <ComposedComponent router={this.context} {...this.props as any} />
-    }
+): React.ComponentType<ExcludeRouterProps<P>> {
+  function WithRouteWrapper(props: any) {
```

Comment:
I feel like this was meant to be called `WithRouterWrapper` instead of `WithRouteWrapper`?